pkg/cryptoauthlib/patches/0001-set-cflags-and-correct-commands.patch

```diff
@@ -0,0 +1,25 @@
+From 9b8457d4f7c351394241739cfd7febf0db59f321 Mon Sep 17 00:00:00 2001
+From: Lena Boeckmann <einhornhool@mobi31.inet.haw-hamburg.de>
+Date: Mon, 4 Nov 2019 13:34:57 +0100
+Subject: [PATCH 1/3] set cflags and correct commands
+
+---
+ lib/CMakeLists.txt | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
+index b15ec66..6973a3c 100644
+--- a/lib/CMakeLists.txt
++++ b/lib/CMakeLists.txt
+@@ -123,7 +124,7 @@ if(ATCA_PRINTF)
+ add_definitions(-DATCAPRINTF)
+ endif(ATCA_PRINTF)
+ 
+-include_directories(cryptoauth PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ../third_party/hidapi/hidapi ${USB_INCLUDE_DIR})
++target_include_directories(cryptoauth PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ../third_party/hidapi/hidapi ${USB_INCLUDE_DIR})
```

Comment:
These are two different ways to add include directories to a target. If I understand correctly they affect different scopes (target scope vs. directory scope). Include_directories adds include paths to all libraries and executables that are defined in the library. This can lead to executables using wrong include files and cause compilation problems.
It doesn't have to be wrong, but in our case there seem to be some issues as the code doesn't compile using the include_directories command. 

Interestingly I found out that I don't need to patch this line in the test's CMakeLists.txt (I did that before and when I removed the patch it still worked).

Also I found several sources saying that target_include_directories should be preferred, because it ist much more specific (only adds include paths to a specific target and not all targets).